File: src/dataset/python/flatten-array_exercism-ThomasZumsteg/claude/ClaudeSonnet4_flatten_array_v3.py
def flatten(iterable):
    if isinstance(iterable, str):
        return list(iterable)
    
    flat = []
    stack = [iterable]
    
    while stack:
        current = stack.pop()
        if isinstance(current, str):
            flat.extend(current)
        elif hasattr(current, '__iter__'):
            try:
                stack.extend(reversed(current))
            except TypeError:
                if current is not None:
                    flat.append(current)
        elif current is not None:
            flat.append(current)
    
    return flat

File: src/dataset/javascript/flatten-array_exercism-ThomasZumsteg/claude/ClaudeSonnet4_flatten-array_v3.js
class Flattener {
    flatten(array) {
        if (array == null) return [];
        if (!Array.isArray(array)) return [array];
        
        const result = [];
        const stack = [array];
        
        while (stack.length > 0) {
            const current = stack.pop();
            
            for (let i = current.length - 1; i >= 0; i--) {
                const element = current[i];
                if (Array.isArray(element)) {
                    stack.push(element);
                } else {
                    result.unshift(element);
                }
            }
        }
        
        return result;
    }
}

module.exports = Flattener;